### PR TITLE
Cassandra set tests

### PR DESCRIPTION
**Pull request by conorbros**

### CODE REVIEW

**Comment by conorbros** [comment]
Update: Trying to find the cause of this failing test, only happens intermittently on my machine.

**Comment by rukai** [comment]
its happening on main branch as well.
I havent been able to reproduce it on my machine.

**Comment by conorbros** [comment]
Doesn't happen when I isolate the `varint` test either...fun!

**Comment by rukai** [comment]
Oh wait, its failing on the test you changed. Ignore what I said.

**Comment by rukai** [comment]
image
maybe you just forgot to implement conversion from Varint to i64?

**Comment by conorbros** [comment]
It sometimes passes on my local machine though

**Comment by rukai** [comment]
Maybe something is failing silently and skipping part of the test under some circumstances?

**Comment by conorbros** [comment]
Alright I've fixed all the issues it seems. Changes: 
- Upstreamed some fixes to cdrs-tokio.
- Increased heap memory for Cassandra. Was running out of memory due to all the data/migrations the test suite does. 
- removed the `duration` type test because it's not actually a native type in Cassandra and is instead part of the custom types.

**Comment by rukai** [comment]
Nice summary of the problems.
I'll take a look now.

File: shotover-proxy/tests/cassandra_int_tests/basic_driver_tests.rs [hunk 1]
```diff
@@ -557,10 +573,176 @@ mod collections {
...
+            }
+
+            // create sets of maps
+            for (i, col_type) in NATIVE_COL_TYPES.iter().enumerate() {
+                run_query(session, format!("CREATE TABLE test_collections_keyspace.test_set_table_map_{} (id int PRIMARY KEY, my_set frozen<set<frozen<map<int, {}>>>>);", i, get_type_str(*col_type)).as_str());
+            }
+        }
+
+        fn insert(session: &Session) {
+            // insert lists of native types
```
Comment (rukai) [inline shotover-proxy/tests/cassandra_int_tests/basic_driver_tests.rs:617]: outdated comment?

File: shotover-proxy/tests/cassandra_int_tests/basic_driver_tests.rs [hunk 2]
```diff
@@ -557,10 +573,176 @@ mod collections {
...
+                        "INSERT INTO test_collections_keyspace.test_set_table_{}_1 (id, my_set) VALUES ({}, {{[{}]}});",
+                        i,
+                        i,
+                        get_type_example(*native_col_type)
+                    )
+                    .as_str(),
+                );
+            }
+
+            fn get_map_example(_key: usize, value: &str) -> String {
```
Comment (rukai) [inline shotover-proxy/tests/cassandra_int_tests/basic_driver_tests.rs:651]: can we deduplicate this?

File: shotover-proxy/tests/cassandra_int_tests/basic_driver_tests.rs [hunk 3]
```diff
@@ -557,10 +573,176 @@ mod collections {
...
+                        i
+                    )
+                    .as_str(),
+                    &[&[ResultValue::Set(vec![ResultValue::List(vec![
+                        get_type_example_result_value(*native_col_type),
+                    ])])]],
+                );
+            }
+
+            // test selecting set  of maps
```
Comment (rukai) [inline shotover-proxy/tests/cassandra_int_tests/basic_driver_tests.rs:717]: double space

**Review by rukai — CHANGES_REQUESTED** [review]
just a few minor changes

File: shotover-proxy/tests/cassandra_int_tests/basic_driver_tests.rs [hunk 4]
```diff
@@ -557,10 +573,176 @@ mod collections {
...
+                    session,
+                    format!(
+                        "CREATE TABLE test_collections_keyspace.test_set_table_{}_{} (id int PRIMARY KEY, my_set frozen<set<{}<{}>>>);",
+                        i,
+                        j,
+                        get_type_str(*collection_col_type),
+                        get_type_str(*native_col_type)
+                    )
+                    .as_str(),
+                );
```
[the remark this answers is not in the document]
Comment (conorbros) [reply shotover-proxy/tests/cassandra_int_tests/basic_driver_tests.rs:618]: Ended up fixing a couple of the other lines in this file aswell

**Review by benbromhead — APPROVED** [review]

**Review by rukai — APPROVED** [review]
LGTM, glad we found the cause of the random failures on main.

**Pull request merged**